src/parser/mod.rs

```diff
@@ -1682,12 +1682,21 @@ impl<'a> Parser<'a> {
     pub fn parse_extract_expr(&mut self) -> Result<Expr, ParserError> {
         self.expect_token(&Token::LParen)?;
         let field = self.parse_date_time_field()?;
-        self.expect_keyword(Keyword::FROM)?;
+        
+        let syntax = if self.parse_keyword(Keyword::FROM) {
+            ExtractSyntax::From
+        } else if self.consume_token(&Token::Comma) {
```

Comment:
The test case covers only snowflake/generic while functionality here is added for all dialects - if the behavior is intended for snowflake, we can add a dialect_of check here? (Otherwise we can move the test to common.rs to cover all dialects)